src/CrawlLogger.php

```diff
@@ -117,49 +118,71 @@ public function crawled(
+        if ($response->getHeader('X-Guzzle-Redirect-History')) {
+            // Retrieve both Redirect History headers
+            $fullRedirectReport = [];
+            // Retrieve both Redirect History headers
+            $redirectUriHistory = $response->getHeader('X-Guzzle-Redirect-History'); // retrieve Redirect URI history
+            $redirectCodeHistory = $response->getHeader('X-Guzzle-Redirect-Status-History'); // retrieve Redirect HTTP Status history
+            // Add the initial URI requested to the (beginning of) URI history
+            array_unshift($redirectUriHistory, (string) $url);
+            // Add the final HTTP status code to the end of HTTP response history
+            array_push($redirectCodeHistory, $response->getStatusCode());
+            $fullRedirectReport = [];
+            foreach ($redirectUriHistory as $key => $value) {
+                $fullRedirectReport[$key] = ['location' => $value, 'code' => $redirectCodeHistory[$key]];
+            }
 
-        if ($this->outputFile && $colorTag === 'error') {
-            file_put_contents($this->outputFile, $message.PHP_EOL, FILE_APPEND);
+            foreach ($fullRedirectReport as $k=>$redirect) {
+                $this->addResult(
+                    (string) $redirect['location'],
+                    (string) $foundOnUrl,
+                    $redirect['code'],
+                    $k + 1 == count($fullRedirectReport) ? $response->getReasonPhrase() : self::REDIRECT
+                );
+            }
+        } else {
+            $this->addResult(
+                (string) $url,
+                (string) $foundOnUrl,
+                $response->getStatusCode(),
+                $response->getReasonPhrase()
+            );
         }
-
-        $this->consoleOutput->writeln("<{$colorTag}>[{$timestamp}] {$message}</{$colorTag}>");
-
-        $this->crawledUrls[$statusCode][] = $url;
     }
 
     public function crawlFailed(
         UriInterface $url,
         RequestException $requestException,
         ?UriInterface $foundOnUrl = null
     ) {
-        $statusCode = $requestException->getResponse()
-            ? $requestException->getResponse()->getStatusCode()
-            : self::UNRESPONSIVE_HOST;
+        if ($response = $requestException->getResponse()) {
+            $this->crawled($url, $response, $foundOnUrl);
+        } else {
+            $this->addResult((string) $url, (string) $foundOnUrl, '---', self::UNRESPONSIVE_HOST);
+        }
+    }
 
-        $reason = $requestException->getResponse()
-            ? $requestException->getResponse()->getReasonPhrase()
-            : $requestException->getMessage();
+    public function addResult($url, $foundOnUrl, $statusCode, $reason)
+    {
+        // done display duplicate results
+        // this happens if a redirect if a redirect is followed to a page
+        if (isset($this->crawledUrls[$statusCode]) && in_array($url, $this->crawledUrls[$statusCode])) {
```

Comment:
Break up this conditional.

Here's a video on that: https://freek.dev/1578-how-to-refactor-complex-if-statements